server/src/main/java/org/apache/cassandra/sidecar/server/MainModule.java

```diff
@@ -691,22 +698,31 @@ public ElectorateMembership electorateMembership(InstancesMetadata instancesMeta
     @Provides
     @Singleton
     public ClusterLeaseClaimTask clusterLeaseClaimTask(Vertx vertx,
+                                                       ServiceConfiguration serviceConfiguration,
                                                        ElectorateMembership electorateMembership,
                                                        SidecarLeaseDatabaseAccessor accessor,
-                                                       ServiceConfiguration serviceConfiguration,
-                                                       PeriodicTaskExecutor periodicTaskExecutor,
                                                        ClusterLease clusterLease,
                                                        SidecarMetrics metrics)
     {
-        ClusterLeaseClaimTask task = new ClusterLeaseClaimTask(vertx,
-                                                               serviceConfiguration,
-                                                               electorateMembership,
-                                                               accessor,
-                                                               clusterLease,
-                                                               metrics);
+        return new ClusterLeaseClaimTask(vertx,
+                                         serviceConfiguration,
+                                         electorateMembership,
+                                         accessor,
+                                         clusterLease,
+                                         metrics);
+    }
+
+    @Provides
+    @Singleton
+    public PeriodicTaskExecutor periodicTaskExecutor(Vertx vertx,
+                                                     ExecutorPools executorPools,
+                                                     ClusterLease clusterLease,
+                                                     ClusterLeaseClaimTask clusterLeaseClaimTask)
+    {
+        PeriodicTaskExecutor periodicTaskExecutor = new PeriodicTaskExecutor(executorPools, clusterLease);
         vertx.eventBus().localConsumer(ON_SIDECAR_SCHEMA_INITIALIZED.address(),
-                                       ignored -> periodicTaskExecutor.schedule(task));
-        return task;
+                                       ignored -> periodicTaskExecutor.schedule(clusterLeaseClaimTask));
```

Comment:
Maybe not for this PR, but I'm not really thrilled that the setup of the `clusterLeaseClaimTask` is in here... I don't think it was really any better before, but somehow starting a periodic task in a provider method like this seems inappropriate - don't really have a simple/sensical answer yet though.